Content.Shared/Strip/ThievingSystem.cs

```diff
@@ -1,17 +1,82 @@
+using Content.Shared.Actions;
+using Content.Shared.Hands.Components;
+using Content.Shared.Interaction.Events;
 using Content.Shared.Inventory;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+using Content.Shared.Popups;
 using Content.Shared.Strip.Components;
 
 namespace Content.Shared.Strip;
 
 public sealed class ThievingSystem : EntitySystem
 {
 
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
     public override void Initialize()
     {
         base.Initialize();
 
         SubscribeLocalEvent<ThievingComponent, BeforeStripEvent>(OnBeforeStrip);
         SubscribeLocalEvent<ThievingComponent, InventoryRelayedEvent<BeforeStripEvent>>((e, c, ev) => OnBeforeStrip(e, c, ev.Args));
+
+        SubscribeLocalEvent<ToggleableThievingComponent, ComponentInit>(OnToggleableThievingInit);
+        SubscribeLocalEvent<ToggleableThievingComponent, ComponentRemove>(OnToggleableThievingRemoved);
+        SubscribeLocalEvent<ToggleableThievingComponent, ToggleThievingActionEvent>(OnToggleThieving);
+
+        SubscribeLocalEvent<ThievingGranterComponent, UseInHandEvent>(OnUseThievingGranter);
+    }
+
+    private void OnUseThievingGranter(EntityUid uid, ThievingGranterComponent component, UseInHandEvent args)
+    {
+        if (HasComp<ToggleableThievingComponent>(args.User))
+        {
+            return;
+        }
+
+        var newComp = EnsureComp<ToggleableThievingComponent>(args.User);
```

Comment:
this should be using a componentregistry and not have anything to do with thieving so other things can reuse it

also a bool for removing after since codespeak manual is reusable and presumably other traitor scrolls

Reply:
Probably best i create an entirely new PR for a skill system - and then convert this to use it.